docs/guides/integration-mobile-client-side.md

```diff
@@ -0,0 +1,700 @@
+
+If the `getAdvertisingToken()` method call returns `null`, there was no identity or valid token generated.
+
+Some possible reasons for this, and some things you could do to troubleshoot, are as follows:
+
+- Check to see whether there are any errors from the `generateIdentity()` call.
+- Enable logging to get more information: see [Enable Logging](#enable-logging)
+- It's possible that the identity that was generated has expired. If this happens, you'll need to call the `generateIdentity()` method again: see [Client-Side Integration Example](#client-side-integration-example).
+
+For more information, see [Best Practices on When to Pass DII to the UID2 Mobile SDK](#best-practices-on-when-to-pass-dii-to-the-uid2-mobile-sdk).
+
+## Best Practices on When to Pass DII to the UID2 Mobile SDK
+
+The first time a new user opens the app, no UID2 identity exists. You'll need to call the `generateIdentity()` method, with the DII, to start the token generation:
+
+<Tabs groupId="language-selection">
+<TabItem value='android' label='Android'>
+
+```js
+UID2Manager.getInstance().generateIdentity()
+```
+
+</TabItem>
+<TabItem value='ios' label='iOS'>
+
+```js
+UID2Manager.shared.generateIdentity()
+```
+
+</TabItem>
+</Tabs>
+
+When this method call completes successfully, the advertising token (UID2 token) is available for you to send to the bid stream.
+
+If the UID2 identity stored in local file storage has expired and cannot be refreshed, you must call the `generateIdentity()` method again to generate a new identity and get the resulting UID2 token.
+
+The only exception is if the following Android method/iOS object returns a status of `OPT_OUT`, which means that the DII has been opted out of UID2 and no identity/token should be generated for it:
+
+<Tabs groupId="language-selection">
+<TabItem value='android' label='Android'>
+
+```js
+UID2Manager.getInstance().getIdentityStatus()
+```
+
+</TabItem>
+<TabItem value='ios' label='iOS'>
+
+```js
+UID2Manager.shared.identityStatus
+```
+
+</TabItem>
+</Tabs>
+
+The best way to determine if DII is required by the UID2 mobile SDKs is to always call the `getAdvertisingToken()` method when the app starts up or resumes:
+
+<Tabs groupId="language-selection">
+<TabItem value='android' label='Android'>
+
+```js
+UID2Manager.getInstance().getAdvertisingToken()
+```
+
+</TabItem>
+<TabItem value='ios' label='iOS'>
+
+```js
+UID2Manager.shared.getAdvertisingToken()
+```
+
+</TabItem>
+</Tabs>
+
+If `getAdvertisingToken()` returns `null`, you'll need to generate a new token. Pass the DII into a call to the `generateIdentity()` method again.
+
+## Opt-Out Handling
+
+(**GWH note: reviewed this doc up to this point. Rest is partially unreviewed (and the below needs fixing).**)
+
+If the DII provided to the `generateIdentity()` method has been opted out of UID2, this method returns `null`. To check the status of the UID2, you can call the following method:
```

Comment:
To check the status of the UID2, you can call the following **_codes_**:

if we say codes maybe it's easier when it's not method call for iOS

Reply:
Just took out "method" here so it's more generic.